File: CS/E1984/Form1.cs
using System;
using System.Data;
using System.Windows.Forms;

namespace E1984 {
    public partial class Form1 : Form
    {
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            tbl.Columns.Add("ParentID", typeof(int));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { string.Format("Name{0}", i), i + 1, 3 - i, DateTime.Now.AddDays(i), i % 3 });
            return tbl;
        }
   

        public Form1()
        {
            InitializeComponent();
            treeList1.DataSource = CreateTable(30);
        }

        private void button1_Click(object sender, EventArgs e)
        {
            MessageBox.Show("Save");
            treeList1.SaveLayoutToXml(MyTreeListSerializeHelper.DefaultFileName);
            MessageBox.Show("Clear");
            treeList1.FormatRules.Clear();
            MessageBox.Show("Restore");
            treeList1.RestoreLayoutFromXml(MyTreeListSerializeHelper.DefaultFileName);
        }  
    }

}

File: VB/E1984/Form1.vb
Imports System
Imports System.Data
Imports System.Windows.Forms

Namespace E1984
    Public Partial Class Form1
        Inherits Form

        Private Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Name", GetType(String))
            tbl.Columns.Add("ID", GetType(Integer))
            tbl.Columns.Add("Number", GetType(Integer))
            tbl.Columns.Add("Date", GetType(Date))
            tbl.Columns.Add("ParentID", GetType(Integer))

            For i = 0 To RowCount - 1
                tbl.Rows.Add(New Object() {String.Format("Name{0}", i), i + 1, 3 - i, Date.Now.AddDays(i), i Mod 3})
            Next

            Return tbl
        End Function

        Public Sub New()
            InitializeComponent()
            treeList1.DataSource = CreateTable(30)
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            MessageBox.Show("Save")
            treeList1.SaveLayoutToXml(DefaultFileName)
            MessageBox.Show("Clear")
            treeList1.FormatRules.Clear()
            MessageBox.Show("Restore")
            treeList1.RestoreLayoutFromXml(DefaultFileName)
        End Sub
    End Class
End Namespace
